src/main/java/com/klodnicki/Bike/v2/service/RentBikeServiceHandler.java

```diff
@@ -98,11 +98,20 @@ public RentResponseDTO rent(RentRequest rentRequest) {
     }
 
     private UserForNormalUserResponseDTO prepareUserDTO (Rent rent) {
+        Set<Authority> authorities = new HashSet<>();
+        for (GrantedAuthority grantedAuthority : rent.getUser().getAuthorities()) {
+            if (grantedAuthority instanceof Authority) { // Authority is a subclass of GrantedAuthority so I check it and then
+                //I convert it to Authority and pass it to builder so that DTO will have appropriate type in authorities
+                //If I don't do this then I get java.lang.ClassCastException: class java.util.LinkedList cannot be cast to
+                // class java.util.Set, is because I was trying to cast a LinkedList to a Set
+                authorities.add((Authority) grantedAuthority);
+            }
+        }
```

Comment:
Okay, so this piece of code required a little bit more work from me as I debugged and checked the whole structure of the project in regard to that.
What I'd do here is in the `UserForNormalUserResponseDTO` I'd place a set, collection or whatever of `AuthorityDTO`. This is the first point here.
Another is that you don't need an `instanceof` here, you don't need this `if` here, you basically can omit all of that, because when you will map those authorities to DTOs, you'll just not need this. Right now what you've created here is just adding layers of unnecessary complexity to the logic. That's why I took a closer look at this :)

By using `getAuthorities()` on User object you get type `Collection<? extends GrantedAuthority>` which should actually be used on the `UserForNormalUserResponseDTO` instead of `Set<Authority>` that you decided to use. But as I stated above, `AuthorityDTO` is a way to go rather that raw entity `Authority`